src/main/java/core/basesyntax/RandomColor.java

```diff
@@ -1,10 +1,18 @@
 package core.basesyntax;
 
+import java.util.function.Supplier;
+
 /**
  * <p>Имплементируйте необходимый функциональный интерфейс и его метод для случайного
  * выбора цвета</p>
  */
-public class RandomColor {
+public class RandomColor implements Supplier<String> {
     private static final String[] COLORS = {"red", "green", "blue", "yellow", "pink",
             "black", "white"};
+
+    @Override
+    public String get() {
+        Supplier color = () -> COLORS[(int)Math.random() * COLORS.length];
+        return (String)color.get();
```

Comment:
you are creating functional interface implementation inside similar functional interface implementation)

Reply:
simplified, thank you.